pkg/sr/api.go

```diff
@@ -348,42 +348,59 @@ func (cl *Client) Schemas(ctx context.Context, subject string) ([]SubjectSchema,
 	return schemas, firstErr
 }
 
-// CreateSchema attempts to create a schema in the given subject.
+// CreateSchemaID attempts to create a schema with a fixed ID and version ID in the given subject,
+// and returns the globally unique identifier assigned by the registry. If the id is set to -1 or 0,
```

Comment:
`If the id is <= 0, the registry...` (and change the logic in the code below where it checks -1 only to handle any negative)

Same thing for the next line: `if the version is <= 0`

I'm trying to make the doc a bit more succinct with these stray details because the functions are getting complicated

Reply:
In the [official documentation](https://docs.confluent.io/platform/current/schema-registry/develop/api.html) it is stated that:

> version (versionId) – Version of the schema to be returned. Valid values for versionId are between [1,2^31-1] or the string latest, which returns the last registered schema under the specified subject. The value -1 is equivalent to latest.

So, technically it does not handle _any negative_ number but specifically the range `[1,2^31-1]` or the value `-1`.

In my opinion, the library should not add any extra semantics on top of the SR API's own semantics, for simplicity. We could add parameter validation or let the API itself error back if a wrong value was sent.  What do you think?

Reply:
Eh I think fine as is, I forgot this was from the official handling.